app/views/dispatch/_messages.html.haml

```diff
@@ -27,8 +29,20 @@
   - title = @conversation.from_phone.phony_formatted(normalize: :US, spaces: '-')
 - else
   - title = "#{h @conversation.user.name} <span>#{@conversation.user.phone_number_normalized.present? ? @conversation.user.phone_number.phony_formatted(normalize: :US, spaces: '-') : ''}</span>".html_safe
-
+  
+- if @conversation&.user&.phone_number_normalized.present?
+  - show_messages = "show();"
+  - no_phone = "hide();"
+- else
+  - show_messages = "hide();"
+  - no_phone = "show();"
+  
 :javascript
+  $('#conversation-messages').#{show_messages}
+  $('.msg-reply').#{show_messages}
+  $('.msg-reply').#{show_messages}
+  $('.no-phone').#{no_phone}
```

Comment:
Would this make more sense as CSS?

```haml
%div.user{:has-phone => @conversation&.user&.phone_number_normalized.present?}
  …
  #conversation-messages
    …
  .msg-reply
    …
  .no-phone
    …
```

```css
.user[has-phone=true] .no-phone,
.user[has-phone=true] #conversation-messages,
.user[has-phone=false] .msg-reply {
  display: none;
}
```

Reply:
The nice thing about the CSS is that it automatically keeps the UI updated as you change DOM attributes, which you can do in HAML or JS.